components/jupyter-web-app/backend/kubeflow_jupyter/common/utils.py

```diff
@@ -387,6 +387,61 @@ def set_notebook_memory(notebook, body, defaults):
     container["resources"]["requests"]["memory"] = memory
 
 
+def set_notebook_gpus(notebook, body, defaults):
+    gpus = None
+    gpuDefaults = defaults.get("gpus", {})
+    if gpuDefaults.get("readOnly", False):
+        # The server should not allow the user to set the GPUs
+        # if the config's value is readOnly. Use the config's value
+        gpus = gpuDefaults["value"]
+        logger.info(f"Using default GPU config: {gpus}")
+
+    elif "gpus" not in body:
+        # Try to load the default values. If they don't exist, don't use GPUs
+        if "gpus" not in defaults:
+            logger.info(
+                "No 'gpus' value in neither the form's body nor in"
```

Comment:
nit: "in neither" -> "in either the form's body or in the default config's values"

Reply:
Thanks, done